control/server.py

```diff
@@ -147,6 +159,14 @@ def _add_server_listener(self):
             self.server.add_insecure_port("{}:{}".format(
                 gateway_addr, gateway_port))
 
+    def _exception_spdk(self):
+        timeout = self.config.getfloat("spdk", "timeout")
+        stdout, stderr = self.spdk_process.communicate(timeout=timeout)
+        return_code = self.spdk_process.returncode
+        self.logger.error(f"SPDK return code: {return_code}")
+        self.logger.error(f"SPDK stdout:\n {stdout}")
+        self.logger.error(f"SPDK stderr:\n {stderr}")
```

Comment:
I believe currently you will always get `None` for stdout and stderr. To read output you would need to add `stderr=subprocess.PIPE, stdout=subprocess.PIPE` params to `subprocess.Popen` when starting the spdk. But we intentionally removed these in efb94c5ebfefd9a61926a778d8e2249039e43e1e, because the spdk may block when it produces a lot of output, which is never read from pipe, and reaches the pipe buffer limit. So currently the spdk just prints to stderr and stdout.